### DISCUSSION TITLE
Is there a built-in way to make LayoutTransformControl affect the Cursor while it is inside it?

### DISCUSSION THREAD

**Discussion by lucdem** [AvaloniaUI/Avalonia#15119] [Q&A]
I have a control the user can zoom into, and I want the cursor to get bigger as the user zooms into it. Basically the same thing as MS Paint's eraser. Besides recreating the cursor bitmap every time the zoom level changes, is there any other way to do this?

**Comment by maxkatz6** [comment]
I am pretty sure ms paint just hides normal cursor, and instead renders eraser rectangle on pointer location. 
Alternatively, you could create bitmaps of different sizes and set custom drawn Cursor from these bitmaps.
I.e. somethign like
```csharp
var bitmap = GetOrCreateBitmapPerSize(currentScaling);
Cursor = new Cursor(bitmap);
```
But it likely to have worse performance and less flexibility, comparing to just drawing rectangle in Avalonia itself.

**Reply by lucdem** [reply]
Hiding the cursor, creating a rectangle at the cursor position and using RenderTransform to make it follow the cursor on PointerMoved events was the first thing I tried, but it wasn't great, there was a very noticeable lag between the actual mouse position and the position of the rectangle, Somewhat similar to what is shown in the video here (https://github.com/AvaloniaUI/Avalonia/discussions/14062). The rectangle also seemed to "skip" frames and jump from one position to the other at times.

Creating a bitmap based on the current zoom level and replacing the cursor with `new Cursor(bitmap)` is what I'm doing right now and it works fine. I was just wondering if there's an easier way to do it. It feels a bit dirty to recreate a new bitmap for the cursor every time the user scrolls (which changes the zoom level when the pointer is inside the control).

**Reply by maxkatz6** [reply]
Well, you can cache created cursors and reuse them. Not sure if it can be done better without going deep into platform code.
